processing/src/main/java/io/druid/segment/virtual/ExpressionColumnValueSelector.java

```diff
@@ -75,4 +75,10 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
     inspector.visit("expression", expression);
     inspector.visit("bindings", bindings);
   }
+
+  @Override
+  public boolean isNull()
+  {
+    return getObject() == null;
```

Comment:
According to the current way how those methods are implemented, it should be `return false`

Reply:
IMO this should be getObject.isNull() instead as the getObject for this class would never return null instead an ExprEval. changing to getObject.isNull()

Reply:
There is an incoherency. This is a selector of `ExprEval`, not of `ExprEval.value()`. This class is used only in `ExpressionSelectors.makeExprEvalSelector()`. This incoherency already causes problems in `AggregatorUtil`: `ExpressionSelectors.makeExprEvalSelector()` is used there only to call `getObject()`, not taking advantage of `ConstantColumnValueSelector` optimization.

To make things more coherent, either
 - `ExpressionSelectors.makeExprEvalSelector()` should return `ColumnValueSelector<T>`, not of `ExprEval` (with according changes)
 - Or `Supplier<ExprEval>` should be returned from this method, enforcing all clients to go via `get().asDouble()` path (and `ConstantColumnValueSelector` should adapt accordingly).

FYI @gianm

Reply:
created https://github.com/druid-io/druid/issues/5406 for follow up. 
This need not block this PR.